control/autoware_autonomous_emergency_braking/src/node.cpp

```diff
@@ -850,7 +895,14 @@ void AEB::getClosestObjectsOnPath(
     obj.distance_to_object = std::abs(dist_ego_to_object);
 
     const Point2d obj_point(p.x, p.y);
-    for (const auto & ego_poly : ego_polys) {
+    // Check if the object is in the predicted path for AEB or not
```

Comment:
@ismetatabay I feel if might be worthwhile to consider using the objects lateral distance instead of their polygons to include and not include them in the speed calculation and classifying them as targets or not. 

if the points absolute lat distance is less than vehicle_width / 2 + expand_width_ they are a target and include them in the object list, if not, we check if their lat distance to the ego path is less than ego_width_ /2 + expand_width + speed_calc_marging. So it is not necessary to iterate through the 2 sets of polygons. The added benefit is that we could use the change in lat distance to calculate object lateral speeds in the future, to "predict" if the object will enter or leave the ego path before a collision.

Reply:
Hello @danielsanchezaran -san,  agree, I updated the PR accordingly. However, I also made changes to the collision checking steps. I merged objects from both the IMU and MPC, and I also merged path points from the IMU and MPC so that the checking mechanism runs only once. What do you think about that?

Reply:
Thank you @ismetatabay I would like to test the PR and see if there is degradation due to the change from polygon-based collision detection to arc coordinate based, I will take a look at processing time. Please give me some time to review. 

Also, I am sorry for being too bothersome, but maybe do you think it is possible to separate this PR into 2? one for the velocity calculation and one for the arc-coordinate based collision detection?